### ISSUE TITLE
IndexError for computing CSA

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#694]
data:
dropbox simon/results/2015-12-14

~~~
sct_process_segmentation -i T2_seg.nii -p csa -t label/template/MNI-Poly-AMU_level.nii.gz -vert 1:1

Check parameters:
.. segmentation file:             T2_seg.nii

Create temporary folder...
mkdir tmp.151214225906_621711/

Copying input data to tmp folder and convert to nii...
sct_convert -i /Users/julien/data/biospective/2015-12-14_sct2.1-2015-12-14/100-011_s2_T2/T2_seg.nii -o tmp.151214225906_621711/segmentation.nii.gz

Change orientation to RPI...
sct_image -i segmentation.nii.gz -setorient RPI -o segmentation_RPI.nii.gz

Open segmentation volume...

Get data dimensions...
  56 x 288 x 288

Smooth centerline/segmentation...
.. Get center of mass of the centerline/segmentation...
.. Smoothing algo = hanning
.. Windows length = 50

Compute CSA...
Traceback (most recent call last):
  File "/Users/julien/code/spinalcordtoolbox/bin/sct_process_segmentation", line 815, in <module>
    main(sys.argv[1:])
  File "/Users/julien/code/spinalcordtoolbox/bin/sct_process_segmentation", line 235, in main
    compute_csa(fname_segmentation, verbose, remove_temp_files, step, smoothing_param, figure_fit, param.file_csa_volume, slices, vert_lev, fname_vertebral_labeling, algo_fitting = param.algo_fitting, type_window= param.type_window, window_length=param.window_length)
  File "/Users/julien/code/spinalcordtoolbox/bin/sct_process_segmentation", line 487, in compute_csa
    normal = normalize(np.array([x_centerline_deriv[iz-min_z_index], y_centerline_deriv[iz-min_z_index], z_centerline_deriv[iz-min_z_index]]))
IndexError: index 278 is out of bounds for axis 0 with size 278
~~~

**Comment by jcohenadad** [comment]
likely related to non-continuous segmentation
<img width="221" alt="screen shot 2015-12-14 at 23 30 38" src="https://cloud.githubusercontent.com/assets/2482071/11796092/bfe4ad64-a2ba-11e5-8f03-704f70e1e7a4.png">
:

**Comment by slevyrosetti** [comment]
`sct_process_segmentation` should not crash in such cases --> 2 potential solutions:
- add a check for non-continuous segmentation (tricky...)
- add an exception in `sct_process_segmentation` to display a warning like " WARNING: Your segmentation is suspicious. Please check it, especially at the extremities."

**Comment by slevyrosetti** [comment]
The issue was addressed in commit #eb8dd71 in branch sl_sct_extract_metric (using the second point described above).

**Comment by slevyrosetti** [comment]
Merged via commit 7f6d9fb.
